automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/NotebookPage.scala

```diff
@@ -179,28 +180,48 @@ class NotebookPage(override val url: String)(override implicit val authToken: Au
     await condition isKernelShutdown
   }
 
+  /**
+    * Throw TimeoutException if Kernel is not ready after restart when timeout is reached
+    *
+    * @param timeout
+    */
   def restartKernel(timeout: FiniteDuration = 1 minute): Unit = {
+    logger.info("restarting kernel ...")
     click on kernelMenu
     click on (await enabled restartKernelSelection)
     click on (await enabled restartKernelConfirmationSelection)
-    await condition (isKernelReady, timeout.toSeconds)
+    await notVisible restartKernelConfirmationSelection
+    await condition (isKernelReady && kernelNotificationText == "none", timeout.toSeconds)
   }
 
-  /**
-    * wait for kernel to become ready. default timeout == 2.minutes
-    */
-  def awaitReadyKernel(timeout: FiniteDuration = 2.minutes): Unit = {
-    await condition (
-      !cellsAreRunning &&
-      find(id("notification_kernel")).exists(_.underlying.getCssValue("display") == "none") &&
-      find(id("kernel_indicator_icon")).exists(_.underlying.getAttribute("class") == "kernel_idle_icon"), timeout.toSeconds
-    )
-    await condition (find(id("kernel_indicator_icon")).exists(_.isDisplayed), 5) // adds little time to check isDisplayed to prevent StaleWebElementException
-  }
+
 
   def clickRunCell(timeout: FiniteDuration = 2.minutes): Unit = {
     click on runCellButton
     awaitReadyKernel(timeout)
   }
 
+  def awaitReadyKernel(timeout: FiniteDuration = 2.minutes): Unit = {
+    val time = Timeout(scaled(Span(timeout.toSeconds, Seconds)))
+    val pollInterval = Interval(scaled(Span(10, Seconds)))
+    eventually(time, pollInterval) {
```

Comment:
Just curious, what's the benefit of using `eventually` over `await condition`?

Also I wonder if `pollInterval` should be configurable by the method caller in addition to `timeout`?

Reply:
I want to use `withClue` enclosed inside `eventually`. Error would be little clearer.